### PR TITLE
refactor(core): Extract deno_core

### PR DESCRIPTION
**Pull request by mmastrac**
`deno_core` is moving out!  You'll find it at https://github.com/denoland/deno_core/ once this PR lands.

### CODE REVIEW

**Review by bartlomieju — CHANGES_REQUESTED** [review]
Marking as request changes so it doesn't get merged by accident

File: cli/tests/integration/check_tests.rs [hunk 1]
```diff
@@ -194,34 +194,34 @@ fn typecheck_declarations_unstable() {
...
+//       "import \"{}\";",
+//       deno_core::resolve_path(
+//         &util::root_path()
+//           .join("core")
+//           .join("lib.deno_core.d.ts")
+//           .to_string(),
+//         &std::env::current_dir().unwrap()
+//       )
+//       .unwrap()
+//     ),
```
Comment (bartlomieju) [inline cli/tests/integration/check_tests.rs:214]: Are you gonna remove this?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/tests/integration/check_tests.rs (hunk 1 above)
Comment (mmastrac) [reply cli/tests/integration/check_tests.rs:214]: I'll get rid of it but I think it needs to find a new home.

**Pull request merged**